tests/integration/ux.test.ts

```diff
@@ -0,0 +1,240 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import fs, { createReadStream } from "fs";
+import minimist from "minimist";
+import readline from "readline";
+import { By, until, WebDriver } from "selenium-webdriver";
+import http from "http";
+
+import {
+  extensionLogsPresent, findAndAct,
+  getChromeDriver,
+  getFirefoxDriver, WAIT_FOR_PROPERTY
+} from "./utils";
+
+
+const args = minimist(process.argv.slice(2));
+console.debug(args);
+for (const arg of ["test_browser", "load_extension", "headless_mode"]) {
+  if (!(arg in args)) {
+    throw new Error(`Missing required option: --${arg}`);
+  }
+}
+
+const testBrowser = args["test_browser"];
+const loadExtension = args["load_extension"] === "true";
+const headlessMode = args["headless_mode"] === "true";
+
+export let webDriver;
+switch (testBrowser) {
+  case "chrome":
+    webDriver = getChromeDriver;
+    break;
+  case "firefox":
+    webDriver = getFirefoxDriver;
+    break;
+  default:
+    throw new Error(`Unknown test_browser: ${testBrowser}`);
+}
+
+console.info(
+  `Running with test_browser: ${testBrowser}, load_extension: ${loadExtension}, headless_mode: ${headlessMode}`
+);
+
+// Wait ten minutes overall before Jest times the test out.
+jest.setTimeout(60 * 10000);
+
+let driver: WebDriver;
+let screenshotCount = 0;
+let server: http.Server;
+
+const PORT = 8000;
+const URL = `http://localhost:${PORT}`;
+const PATH = "webarchive/localhost";
+
+const waitForExtensionLog = async (message) => {
+  await driver.wait(
+    async () =>
+      await extensionLogsPresent(
+        driver,
+        testBrowser,
+        message
+      ),
+    WAIT_FOR_PROPERTY
+  );
+}
+
+describe("WebScience Test Extension", function () {
+  beforeAll(async () => {
+    server = http.createServer((req, res) => {
```

Comment:
Not sure if we are planning to add more complex pages to the webarchive, but if so I think it makes sense to switch to something like the below to allow for folder structure.
```
// npm i -D http-server
import { spawn } from "child_process";
let http-server;
...
beforeAll(() => {
    http-server = spawn("http-server", PATH, "-p", PORT);
}

afterAll(() => {
    http-server.kill();
}

Reply:
Yes I think we will, one thing we've been working on elsewhere is making it possible for folks to build their own local archive for test purposes: https://github.com/mozilla-rally/facebook-pixel-hunt/blob/main/tests/integration/webarchive/create-archive.sh

I think using a webserver that supports static files is a good idea, I was trying to avoid dragging in dependencies for thhis first update but I think it's worth it.